packages/scenes/src/querying/ExtraQueryProvider.ts

```diff
@@ -0,0 +1,46 @@
+import { DataQueryRequest, PanelData } from "@grafana/data";
+
+import { SceneObjectBase } from "../core/SceneObjectBase";
+import { SceneObjectState } from "../core/types";
+
+// A processor function called by the query runner with responses
+// to any extra requests.
+//
+// A processor function should accept two arguments: the data returned by the
+// _primary_ query, and the data returned by the `ExtraQueryProvider`'s
+// _secondary_ query. It should return a new `PanelData` representing the processed output.
+// It should _not_ modify the primary PanelData.
+//
+// Examples of valid processing include alignment of data between primary and secondary
+// (see the `timeShiftAlignmentProcessor` returned by `SceneTimeRangeCompare`), or doing
+// some more advanced processing such as fitting a time series model on the secondary data.
+//
+// See the docs for `extraQueryProcessingOperator` for more information.
+export type ExtraQueryDataProcessor = (primary: PanelData, secondary: PanelData) => PanelData;
```

Comment:
Don't think it's suitable for this PR but if I want to offload my processing to a web worker this is going to need to return a Promise instead...

```suggestion
export type ExtraQueryDataProcessor = (primary: PanelData, secondary: PanelData) => Promise<PanelData>;
```

Reply:
If we don't introduce this now, it's going to be a breaking change in the future. I think it may be worth doing this straight away, wdyt @torkelo ?

Reply:
I was thinking we _could_ change it to `export type ExtraQueryDataProcessor = (primary: PanelData, secondary: PanelData) => PanelData | Promise<PanelData>;` but it is clunky, I agree...

Reply:
Can it be an observable (like the current post processing) that will be much easier to slot in to the rxjs pipeline

Reply:
yeah, +1 for that!

Reply:
I've changed the type to `export type ExtraQueryDataProcessor = (primary: PanelData, secondary: PanelData) => Observable<PanelData>;` in bdab9f4d8dfdc413887d3a15ef1ba32210399603, is that what you meant? :thinking: Think I got the types right but I might have misunderstood.

Reply:
i think that's exactly it @sd2k